src/components/structures/auth/forgot-password/EnterEmail.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useRef } from "react";
+
+import { Icon as EMailIcon } from "../../../../../res/img/element-icons/Email-icon.svg";
+import { _t, _td } from '../../../../languageHandler';
+import EmailField from "../../../views/auth/EmailField";
+import { ErrorMessage } from "../../ErrorMessage";
+import Spinner from "../../../views/elements/Spinner";
+import Field from "../../../views/elements/Field";
+
+interface EnterEmailProps {
+    email: string;
+    errorText: string | null;
+    loading: boolean;
+    onInputChanged: (stateKey: string, ev: React.FormEvent<HTMLInputElement>) => void;
+    onSubmitForm: (ev: React.FormEvent) => void;
+}
+
+/**
+ * This component renders the email input view of the forgot password flow.
+ */
+export const EnterEmail: React.FC<EnterEmailProps> = ({
+    email,
+    errorText,
+    loading,
+    onInputChanged,
+    onSubmitForm,
+}) => {
+    const submitButtonChild = loading
+        ? <Spinner w={16} h={16} />
+        : _t("Send email");
+
+    const emailFieldRef = useRef<Field>(null);
+
+    const onSubmit = async (event: React.FormEvent) => {
+        if (await emailFieldRef.current?.validate({ allowEmpty: false })) {
+            onSubmitForm(event);
+            return;
+        }
+
+        emailFieldRef.current?.focus();
+        emailFieldRef.current?.validate({ allowEmpty: false, focused: true });
+    };
+
+    return <>
+        <EMailIcon className="mx_AuthBody_icon" />
+        <h1>{ _t("Enter your email to reset password") }</h1>
+        <p className="mx_AuthBody_text">
+            {
+                _t(
+                    "<b>%(homeserver)s</b> will send you a verification link to let you reset your password.",
+                    { homeserver: "matrix.org" },
```

Comment:
This shouldn't be hardcoded?

Reply:
Oh, yes sure. Matrix.org is THE homeserver :upside_down_face: 

Good catch :+1: Fixing this and update the tests…